src/Mutator/MutatorFactory.php

```diff
@@ -82,33 +76,16 @@ public function create(array $resolvedMutators): array
             /** @var string[] $ignored */
             $ignored = $config['ignore'] ?? [];
 
-            switch ($mutatorClass) {
-                case BCMath::class:
-                    $mutator = new BCMath(new BCMathConfig($settings));
+            if (in_array(ConfigurableMutator::class, class_implements($mutatorClassName), true)) {
```

Comment:
Why not use an instance of check here?

Reply:
If we use this interface it should also declare the constructor.

Reply:
>Why not use an instance of check here?

Could probably do the job, dunno might have mixed things up with the test provider 😅 

>If we use this interface it should also declare the constructor.

Why?

Reply:
Because we rely on the fact that the constructor takes The config class as a param.

Reply:
This won't work https://3v4l.org/407tp

You could keep `MutatorConfig` but then we would loose all the benefits of having a specific config... Also as of now this is guaranteed by the tests which uses on the factory

Another alternative is too keep `__construct(array $settings)` in `Mutator` and just instantiate the config directly from the mutator...

Reply:
>Why not use an instance of check here?

Ha obviously: because we don't have the mutator instance yet, just its class name. So `instanceof` won't work there